app/models/concerns/host_searching.rb

```diff
@@ -87,6 +90,10 @@ def filter_with_results_for_policy(_filter, _operator, policy_or_profile_id)
     end
 
     def filter_by_policy(_filter, _operator, policy_or_profile_id)
+      profiles = profiles_by_id(policy_or_profile_id).to_a.push(Profile.new)
```

Comment:
@skateman This did end up working but I had to do `to_a` before I pushed the new profile(apparently pushing directly onto an activerecord relation was possible before Rails 5 but now you have to convert it to an array). I also had to move the set up of `profiles` into a scope to fix a few rubocop issues. 
Does this look good to you? I left the other solution in `with_benchmark` in case we wanted to go back to that one but I'll remove that if we decide this new solution looks good.

Reply:
I looked into the `ActiveRecord::Relation` and as it is an `Enumerable` it inherently supports the `+` operation, which probably also converts it to an array, but it seems more elegant and readable. I'm not sure if it will like a `+` with a `Profile` class, but you could have another relation to work with by tricking rails of creating a record with `nil` as its ID. The simplest way I think is `Profile.select('NULL AS id').distinct` that should return you a relation having exactly one profile with `nil` as its id. I am 100% sure you could do a `+` operation between two relations :wink: 

Regarding the scope, I'd rather have it as a method, ideally a private one, but maybe after my proposal, Rubocop will stop complaining.

Reply:
The `+` operation worked well with `Profile.select('NULL AS id').distinct` 👍  I also moved the set up of `profiles` into a private method instead of a scope.